apps/examples/src/examples/speech-bubble/SpeechBubble/SpeechBubbleUtil.tsx

```diff
@@ -97,60 +98,68 @@ export class SpeechBubbleUtil extends ShapeUtil<SpeechBubbleShape> {
 	}
 
 	override getHandles(shape: SpeechBubbleShape) {
-		const handles = getHandleinShapeSpace(shape).props.handles
-		const handlesArray = Object.values(handles)
-
-		return handlesArray
+		const {
+			handles: { handle },
+			w,
+			h,
+		} = shape.props
+
+		return [
+			{
+				...handle,
+				// props.handles.handle coordinates are normalized
+				// but here we need them in shape space
+				x: handle.x * w,
+				y: handle.y * h,
+			},
+		]
 	}
 
 	// [4]
 	override onBeforeUpdate: TLOnBeforeUpdateHandler<SpeechBubbleShape> | undefined = (
 		_: SpeechBubbleShape,
 		next: SpeechBubbleShape
 	) => {
-		const shape = getHandleinShapeSpace(next)
+		const { w, h, handles } = next.props
 
-		const { originalIntersection: intersection, insideShape } = getHandleIntersectionPoint({
-			w: shape.props.w,
-			h: shape.props.h,
-			handle: shape.props.handles.handle,
-		})
-
-		if (!intersection) throw new Error('No intersection')
+		const { segmentsIntersection, insideShape } = getHandleIntersectionPoint(next)
 
-		const intersectionVector = new Vec2d(intersection.x, intersection.y)
-		const handleVector = new Vec2d(shape.props.handles.handle.x, shape.props.handles.handle.y)
+		const slantedLength = Math.hypot(w, h)
+		const MIN_DISTANCE = slantedLength / 5
+		const MAX_DISTANCE = slantedLength / 1.5
 
-		const topLeft = new Vec2d(0, 0)
-		const bottomRight = new Vec2d(shape.props.w, shape.props.h)
-		const center = new Vec2d(shape.props.w / 2, shape.props.h / 2)
-		const MIN_DISTANCE = topLeft.dist(bottomRight) / 5
+		const handleInShapeSpace = new Vec2d(handles.handle.x * w, handles.handle.y * h)
 
-		const MAX_DISTANCE = topLeft.dist(bottomRight) / 1.5
+		const distanceToIntersection = handleInShapeSpace.dist(segmentsIntersection)
+		const center = new Vec2d(w / 2, h / 2)
+		const vHandle = Vec2d.Sub(handleInShapeSpace, center).uni()
 
-		const distanceToIntersection = handleVector.dist(intersectionVector)
-		const angle = Math.atan2(handleVector.y - center.y, handleVector.x - center.x)
-		let newPoint = handleVector
```

Comment:
It's confusing because we use Vec2d for both points and vectors, but when naming stuff remember that "vector" is used to indicate a "direction" or 2d distance, while a "point" is used to indicate a "place" (technically a 2d distance from the coordinate space's origin to a point).